File: src/main/java/br/org/acal/resouces/adapter/CustomerAdapter.java
package br.org.acal.resouces.adapter;

import br.org.acal.commons.StringUtil;
import br.org.acal.domain.entity.Customer;
import br.org.acal.domain.entity.Document;
import br.org.acal.resouces.model.CustomerModel;
import lombok.val;

public class CustomerAdapter {
    public static Customer map(CustomerModel customerModel){
        return Customer
            .builder()
            .number(customerModel.getNumber())
            .name(customerModel.getName().trim())
            .document(Document.builder()
                .type(getDocumentType(customerModel))
                .number(getCleanDocumentNumber(customerModel))
                .build())
            .build();
    }

    private static String getCleanDocumentNumber(CustomerModel customerModel){
        return StringUtil.clean(customerModel.getCpf() == null ? customerModel.getCnpj() : customerModel.getCpf());
    }

    private static String getDocumentType(CustomerModel customerModel){
        return customerModel.getCpf() == null ? "CNPJ" : "CPF";
    }

    public static CustomerModel map(Customer customer){
        return CustomerModel
            .builder()
                .number(customer.getNumber())
                .name(customer.getName())
            .build();
    }

}

File: src/main/kotlin/br/org/acal/resources/adapter/CustomerAdapter.kt
package br.org.acal.resources.adapter

import br.org.acal.domain.entity.Customer
import br.org.acal.domain.entity.Document
import br.org.acal.domain.entity.Partner
import br.org.acal.resources.model.CustomerModel

class CustomerAdapter {
    fun map(customerModel: CustomerModel): Customer {

        val type = customerModel.cpf?.let { "CPF" } ?: "CNPJ"
        val number = customerModel.cpf?.let { "CPF" } ?: "CNPJ"

        return Customer(
            number = customerModel.number,
            name = "${customerModel.name.trim()} ${customerModel.lastName?.trim()}",
            document = Document(type, number),
            partner = Partner(customerModel.partner)
        )
    }

    fun map(customer: Customer): CustomerModel {
        return when (customer.document.type){
            "CPF" -> CustomerModel(name = customer.name, number = customer.number, cpf = customer.document.number)
            else -> CustomerModel(name = customer.name, number = customer.number, cnpj = customer.document.number)
        }
    }
}
